git-bash-hints.qmd

```diff
@@ -0,0 +1,188 @@
+
+`pwd`: This stands for "print working directory", so it will show you your current working directory for Git.
+
+You can use it by entering the following in the terminal: 
+
+```{r, eval = FALSE}
+
+pwd
+
+```
+
+
+`cd`: This stands for "change directory", so it will point to where you want your directory to be.
+
+To change your working directory, we would amend the code below to reflect the path of the working directory we want and enter it into the terminal: 
+
+```{r, eval = FALSE}
+
+cd /path_to/working_directory/you_want
+
+```
+
+ 
+## Viewing Files and Directories
+
+The commands in this section help you list and view the contents of directories, providing detailed information about files and sub-directories.
+
+`ls`: This stands for "listing".
+
+To list all files or sub-directories, we would type the following in the terminal: 
+
+```{r, eval = FALSE}
+
+ls
+
+```
+
+
+To list all files or sub-directories, including hidden files, we would type the following in the terminal: 
+
+```{r, eval=FALSE}
+
+ls -a
+
+```
+
+
+## Creating and Removing Files and Directories
+
+### Creating directories 
+
+The commands in this section enable you to create new directories, remove files, and delete empty directories.
+
+
+`mkdir`: This stands for "make directory".
+
+To create a directory, we would amend the code below to reflect the path we want and the new directory name and enter it into the terminal: 
+
+```{r, eval=FALSE}
+
+mkdir /path_to_the_directory_you_want_to_create/new_directory_name
+
+```
+
+### Removing directories 
+
+:::{.callout-important}
+
+Make sure you're in the correct directory/using the correct path before using commands in this section.
+:::
+
+`rm`: This stands for "remove" and we can use it to remove files and directories.
+
+To remove a directory and its contents, we would amend the code below to reflect the path and the directory we want and enter it into the terminal: 
+
+
+```{r, eval=FALSE}
+
+rm -r /path_to_the_directory/directory_name
+
+```
+
+### Removing files
+
+ 
+Assuming we're already in the directory we want, we can remove a file from it by amending the code below to reflect the file name we want to remove, and its extension and then enter it into the terminal: 
+
+
+```{r, eval=FALSE}
+
+rm -r file_name.extention
+
+```
+
+You need to make sure to include the extension of the file in the command, or it will not work. 
+
+## Viewing File Contents
+
+:::{.callout-important}
+
+Make sure you're in the correct directory/using the correct path before using commands in this section.
+:::
+
+The commands in this section allow you to display the contents of files, either in full or in parts, for easy reading and inspection.
+
+To display the first 10 lines of a file, we would amend the code below to reflect the file name we want to view and its extension and then enter it into the terminal:
+
+
+```{r, eval=FALSE}
+
+head file_name.extention
```

Comment:
Useful thing to mention here and in tail is that you can set the number of lines to return using e.g.:
`tail -3 filename`
and
`head -12 filename`